FRAuth/FRAuth/Callbacks/DeviceSigningVerifierCallback.swift

```diff
@@ -0,0 +1,243 @@
+        
+        self.timeout = outputDictionary[CBConstants.timeout] as? Int
+        
+        //parse inputs
+        var inputNames = [String]()
+        for input in inputs {
+            guard let inputName = input[CBConstants.name] as? String else {
+                throw AuthError.invalidCallbackResponse("Failed to parse input")
+            }
+            inputNames.append(inputName)
+        }
+        
+        guard let jwsKey = inputNames.filter({ $0.contains(CBConstants.jws) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing jwsKey")
+        }
+        self.jwsKey = jwsKey
+        
+        guard let clientErrorKey = inputNames.filter({ $0.contains(CBConstants.clientError) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing clientErrorKey")
+        }
+        self.clientErrorKey = clientErrorKey
+        
+        try super.init(json: json)
+        type = callbackType
+        response = json
+    }
+    
+    
+    /// Sign the device.
+    /// - Parameter completion Completion block for Device binding result callback
+    open func sign(completion: @escaping DeviceBindingResultCallback) {
+        execute(userKeyService: nil, completion)
+    }
+    
+    
+    /// Helper method to execute binding , signing, show biometric prompt.
+    /// - Parameter userKeyService: service to sort and fetch the keys stored in the device
+    /// - Parameter completion: completion Completion block for Device binding result callback
+    internal func execute(userKeyService: UserKeyService?,
+                          _ completion: @escaping DeviceBindingResultCallback) {
+        let newUserKeyService = userKeyService ?? UserDeviceKeyService(encryptedPreference: nil)
+        
+        let status = newUserKeyService.getKeyStatus(userId: userId)
+        
+        switch status {
+        case .singleKeyFound(key: let key):
+            authenticate(userKey: key, authInterface: nil, completion)
+        case .multipleKeysFound(keys: _):
+            getUserKey(userKeyService: newUserKeyService) { key in
+                if let key = key {
+                    self.authenticate(userKey: key, authInterface: nil, completion)
+                } else {
+                    self.handleException(status: .abort, completion: completion)
+                }
+            }
+            break
+        case .noKeysFound:
+            handleException(status: .unRegister, completion: completion)
+        }
+    }
+    
+    
+    /// Helper method to execute signing, show biometric prompt.
+    /// - Parameter userKey: User Information
+    /// - Parameter authInterface: Interface to find the Authentication Type - provide nil to default to getDeviceBindingAuthenticator()
+    /// - Parameter completion: completion Completion block for Device binding result callback
+    internal func authenticate(userKey: UserKey,
+                               authInterface: DeviceAuthenticator?,
```

Comment:
The DeviceAuthenticator has struct BiometricOnly,None, DevicePin.....Change to a class instead of struct, because we may need to inherit those classes which will not supported by struct